### DISCUSSION TITLE
ESP8266 resetting when testing code from Thonny

### DISCUSSION THREAD

**Discussion by shantanudasgupta** [micropython/micropython#9771] [General]
Hi,
I am testing scripts written in While loop in Thonny in a NodeMCU(ESP8266) module. Occassionally the ESP resets without any reason. I suspect this is a problem with the PC power supply which is used to power the ESP board since the timing of reset is not regular.
There is no auto-reset in Thonny...I checked.
Can anyone throw some light into this?
Thanks.

**Comment by robert-hh** [comment]
What do you get as reset cause when the device restarts? You get that value with:

```
import machine
print(machine.reset_cause())
```
Just add these lines into boot.py or main.py.

**Reply by shantanudasgupta** [reply]
Thanks Robert.
Actually there is no main.py....I'm just testing a script by F5.So I cannot verify the reset cause. When there is a reset the micropython REPL prompt appears. In watchdog reset the cause is printed out in the REPL....I've seen that. So this must be a power spike/hardware reset.
I'll check the USB settings in Windows 10 ....whether the low power mode is activated and Windows withdraws power from the port. But this is unlikely since the +5V is there always ....I've checked by multimeter.

**Comment by shantanudasgupta** [comment]
Disabled USB Selective suspend mode & put on trial once again

**Comment by shantanudasgupta** [comment]
Pronlem solved by disabling Windows 10 USB selective suspend.
